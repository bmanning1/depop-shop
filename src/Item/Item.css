.item {
    margin: 0.5rem;
    width: 15rem;
}

.item p {
    margin: 0.5rem 0;
    font-size: 0.8rem;
}

.item-img {
    display: block;
    background-color: rgb(215, 215, 215);
    width: 100%;
}

.item-img-container {
    position: relative;
    width: 15rem;
    height: 15rem;
    overflow: hidden;
}

.sold {
    position: absolute;
    top: 0;
    background: rgba(0, 0, 0, 0.5);
    width: 100%;
    height: 100%;
    color: white;
    display: flex;
    justify-content: center;
    align-items: center;
    font-size: 3rem;
}

.like-button {
    position: absolute;
    bottom: 0.5rem;
    right: 0.5rem;
    background: none;
    border: none;
    cursor: pointer;
    padding-top: 0.2rem;
    border-radius: 100%;
    height: 2rem;
    width: 2rem;
    display: flex;
    justify-content: center;
}

.like-button:focus {
    outline: none;
}

.heart {
    transition: 0.3s;
    font-size: 1.2rem;
    color: white;
}

.liked *, .heart:hover {
    color: red;
}
