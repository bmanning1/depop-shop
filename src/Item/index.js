import React from 'react';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faHeart } from '@fortawesome/free-solid-svg-icons'
import './Item.css';

const Item = ({ id, img, title, size, price, seller, sold, brand, liked, toggleLike }) => (
  <div className="item">
    <div className="item-img-container">
      <img className="item-img" alt={title} src={img} />
      {sold && <div className="sold">SOLD</div>}
      <button className={`like-button ${liked ? 'liked' : ''}`} onClick={() => toggleLike(id, title)}>
        <FontAwesomeIcon className="heart" icon={faHeart} />
      </button>
    </div>
    <p><b>{seller}</b> {title}</p>
    <p>{brand && <span>Brand: <b>{brand}</b></span>} {size && <span>Size: <b>{size}</b></span>}</p>
    <p>Price <b>{`Â£${price}`}</b></p>
  </div>
);

Item.propTypes = {
  id: PropTypes.number.isRequired,
  img: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
  seller: PropTypes.string.isRequired,
  sold: PropTypes.bool.isRequired,
  size: PropTypes.string,
  brand: PropTypes.string,
  liked: PropTypes.bool.isRequired,
  toggleLike: PropTypes.func.isRequired
}

Item.defaultProps = {
  size: null,
  brand: null
}

export default Item;