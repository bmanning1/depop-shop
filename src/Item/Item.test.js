import React from 'react';
import { shallow } from 'enzyme';
import Item from '.';

describe('Item:', () => {
    let baseComponent; 
    const props = {
        id: 1,
        img: 'http://www.imgUrl.com',
        title: 'Jumper',
        price: 35.50,
        seller: '@user',
        sold: false,
        liked: false,
        toggleLike: jest.fn()
    };

    beforeEach(() => {
        baseComponent = shallow(<Item {...props} />);
    });

    test('should match snapshot', () => {
        expect(baseComponent).toMatchSnapshot();
    });

    test('should display sold over the image if the item is sold', () => {
        const newProps = { ...props, sold: true };
        const component = shallow(<Item {...newProps} />);
        expect(baseComponent).toMatchDiffSnapshot(component);
    });

    test('should add liked styling on the heart icon if the item is liked', () => {
        const newProps = { ...props, liked: true };
        const component = shallow(<Item {...newProps} />);
        expect(baseComponent).toMatchDiffSnapshot(component);
    });

    test('should display size and brand if they are given', () => {
        const newProps = { ...props, size: 'L', brand: 'Supreme' };
        const component = shallow(<Item {...newProps} />);
        expect(baseComponent).toMatchDiffSnapshot(component);
    });
});