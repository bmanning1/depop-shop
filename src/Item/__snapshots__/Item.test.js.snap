// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item: should add liked styling on the heart icon if the item is liked 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -8,11 +8,11 @@
        alt=\\"Jumper\\"
        className=\\"item-img\\"
        src=\\"http://www.imgUrl.com\\"
      />
      <button
-       className=\\"like-button \\"
+       className=\\"like-button liked\\"
        onClick={[Function onClick]}
      >
        <FontAwesomeIcon
          border={false}
          className=\\"heart\\""
`;

exports[`Item: should display size and brand if they are given 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -52,11 +52,23 @@
      </b>
       
      Jumper
    </p>
    <p>
+     <span>
+       Brand: 
+       <b>
+         Supreme
+       </b>
+     </span>
       
+     <span>
+       Size: 
+       <b>
+         L
+       </b>
+     </span>
    </p>
    <p>
      Price 
      <b>
        £35.5"
`;

exports[`Item: should display sold over the image if the item is sold 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -7,10 +7,15 @@
      <img
        alt=\\"Jumper\\"
        className=\\"item-img\\"
        src=\\"http://www.imgUrl.com\\"
      />
+     <div
+       className=\\"sold\\"
+     >
+       SOLD
+     </div>
      <button
        className=\\"like-button \\"
        onClick={[Function onClick]}
      >
        <FontAwesomeIcon"
`;

exports[`Item: should match snapshot 1`] = `
<div
  className="item"
>
  <div
    className="item-img-container"
  >
    <img
      alt="Jumper"
      className="item-img"
      src="http://www.imgUrl.com"
    />
    <button
      className="like-button "
      onClick={[Function]}
    >
      <FontAwesomeIcon
        border={false}
        className="heart"
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              512,
              512,
              Array [],
              "f004",
              "M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z",
            ],
            "iconName": "heart",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
    </button>
  </div>
  <p>
    <b>
      @user
    </b>
     
    Jumper
  </p>
  <p>
     
  </p>
  <p>
    Price 
    <b>
      £35.5
    </b>
  </p>
</div>
`;
